### PR TITLE
contact form

### LINKED ISSUE #147: Feedback Form for GUI
Add a form where the user can submit feedback (an email is generated)

Spam protection should be provided

the exact URL should be sent so it will be easier to find out what the reported problem is about

### PR DESCRIPTION
**Pull request by tobiasschweizer**
This PR adds a contact form component. The form's content is stored in Cloud Firestore.

closes #147

### CODE REVIEW

File: src/environments/environment.ts [hunk 1]
```diff
@@ -26,5 +26,13 @@ export const environment = {
     api: 'http://localhost:3333',           // perhaps we have to rename it into knora?
     url: 'http://localhost:4200',           // perhaps we have to rename it into guiUrl or salsah?
     localData: 'data',
-    pagingLimit: 25
+    pagingLimit: 25,
+    firebase: {
```
Comment (tobiasschweizer) [inline src/environments/environment.ts]: testing only. the credentials shouldn't be published ...

**Comment by subotic** [comment]
Isn't this against the policy of DaSCH and SAGW to store data outside of Switzerland? Is there any technical reason why this data couldn't be stored inside the triplestore via https://github.com/dhlab-basel/Knora/issues/524?

**Comment by tobiasschweizer** [comment]
@subotic 

> Isn't this against the policy of DaSCH and SAGW to store data outside of Switzerland?

I have also thought of this but the solution at hand seems the simplest. I do not think that the content submitted via the contact form should end up in the triplestore. The idea is that people could submit reports of problems and bugs they experienced.

If you can think of a simple alternative on the production server (e.g., emails sent by the webserver), we do not have to use a third-party database.

**Comment by subotic** [comment]
It would end into the Key-Value store, which happens to be built uppon the triplestore for simplicity. Alternatively we would need to run a third-party KV-Store, which would add more complexity to the deployment.

The GUI and also other parts of Knora will use this KV-Store anyway. If you like, you can also use it for the feedback form data.

We will be able to send emails from the server, but then someone would need to receive them and answer.

**Comment by tobiasschweizer** [comment]
> The GUI and also other parts of Knora will use this KV-Store anyway. If you like, you can also use it for the feedback form data.

Ok! What's the route for that? And how would we handle spam protection?

**Comment by subotic** [comment]
This would be the open issue: https://github.com/dhlab-basel/Knora/issues/524

> And how would we handle spam protection?

Maybe by adding a captcha to requests which come from unauthorized, i.e. not logged-in users?

**Pull request merged**